ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/ApiSecurityContext.java

```diff
@@ -16,11 +16,15 @@
 package io.confluent.ksql.api.auth;
 
 import io.confluent.ksql.security.KsqlPrincipal;
+import java.util.List;
+import java.util.Map.Entry;
 import java.util.Optional;
 
 public interface ApiSecurityContext {
 
   Optional<KsqlPrincipal> getPrincipal();
 
   Optional<String> getAuthToken();
+
+  List<Entry<String, String>> getRequestHeaders();
```

Comment:
Passing incoming request headers as part of the ApiSecurityContext in this way means the existing `Optional<String> getAuthToken();` is not strictly needed anymore since the auth token is extracted from the auth header. I've decided to leave it in despite the redundancy since the auth token is used in multiple places (both the connect client and the internal ksql client) and passing an extra string seems fine compared to having code in multiple places for extracting the header. Happy to refactor if others disagree, though.